dev-tools/xtask/src/a4x2/a4x2_deploy.rs

```diff
@@ -0,0 +1,938 @@
+pub struct A4x2DeployArgs {
+    #[command(subcommand)]
+    command: DeployCommand,
+}
+
+#[derive(Subcommand, Clone)]
+pub enum DeployCommand {
+    /// Download and install the propolis hypervisor and bootcode. Pre-requisite
+    /// for launching a4x2.
+    InstallPropolis,
+
+    /// Start a4x2, deploy a control plane to it, and then leave it running.
+    /// Destroys any existing a4x2 deployment, unless --ensure is used.
+    Start(StartArgs),
+
+    /// Stop a4x2 that was previously launched with the `start` subcommand.
+    Stop,
+
+    /// Execute live-tests previously built by `xtask a4x2 package`.
+    RunLiveTests,
+
+    /// Collect logs from a4x2, and services within the sleds.
+    CollectEvidence,
+
+    /// Query a4x2 node access information
+    Status,
+}
+
+#[derive(Args, Clone)]
+pub struct StartArgs {
+    /// Path to an a4x2-package bundle generated by a4x2-package xtask.
+    #[clap(long, default_value_t = Utf8PathBuf::from(super::DEFAULT_A4X2_PKG_PATH))]
+    package: Utf8PathBuf,
+
+    /// Makes the command idempotent. By default, `start` will destroy any
+    /// existing a4x2 deployment for easy development iteration. With --ensure,
+    /// the command will *ensure* a4x2 is running, starting it if needed, but
+    /// leaving an existing deployment alone.
+    #[clap(long)]
+    ensure: bool,
+}
+
+struct Environment {
+    /// Directory within which we will unpack the a4x2 package and launch a4x2
+    work_dir: Utf8PathBuf,
+
+    /// Directory in `work_dir` containing a4x2
+    a4x2_dir: Utf8PathBuf,
+
+    /// Falcon state directory
+    falcon_dir: Utf8PathBuf,
+
+    /// Directory in `work_dir` where logs are written by collect-evidence
+    out_dir: Utf8PathBuf,
+}
+pub fn run_cmd(args: A4x2DeployArgs) -> Result<()> {
+    let sh = Shell::new()?;
+
+    let env = {
+        let pwd = Utf8PathBuf::try_from(env::current_dir()?)?;
+        let work_dir = pwd.join("target/a4x2/deploy");
+
+        // a4x2 dir. will be created by the unpack step
+        let a4x2_dir = work_dir.join(super::A4X2_PACKAGE_DIR_NAME);
+
+        // falcon dir. created by a4x2
+        let falcon_dir = a4x2_dir.join(".falcon");
+
+        // Output logs go here
+        let out_dir = work_dir.join("output-logs");
+
+        Environment { work_dir, a4x2_dir, falcon_dir, out_dir }
+    };
+
+    match args.command {
+        DeployCommand::Start(args) => {
+            // Teardown previous deploy if it exists, before wiping the data
+            // for it.
+            if env.falcon_dir.try_exists()? {
+                if args.ensure {
+                    // ensure arg says: make sure a4x2 is in theory deployed.
+                    // right now we will use the presence of .falcon to indicate
+                    // that, but that can be an incorrect if the system rebooted
+                    // since a4x2 was deployed.
+                    eprintln!(".falcon directory exists; assuming a4x2 is up");
+                    return Ok(());
```

Comment:
It seems like this isn't reliable in that it will only _not_ tear down if the falcon directory is present.  But it might be present, yet nothing is running.  I wonder: what's the motivation for this flag?